casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -300,10 +300,11 @@ object BlockAPI {
 
     def casperResponse(implicit casper: MultiParentCasper[F]): F[ApiErr[A]] =
       for {
-        dag      <- MultiParentCasper[F].blockDag
-        depth    <- maybeDepth.fold(dag.topoSort(0L).map(_.length - 1))(_.pure[F])
-        topoSort <- dag.topoSortTail(depth)
-        result   <- doIt(casper, topoSort)
+        dag              <- MultiParentCasper[F].blockDag
+        totalBlockNumber <- dag.topoSort(0L, none).map(_.length)
+        depth            = maybeDepth.getOrElse(totalBlockNumber)
+        topoSort         <- dag.topoSort((totalBlockNumber - depth).toLong, none)
```

Comment:
Even if `depth` value is specified `topoSort` will be called with start `0`, which will cause load of all block hashes. 

Previous code is also not correct, it calls `topoSort(0)` two times when depth is not specified. With all that, what is real use case for depth optional?